object-api.go

```diff
@@ -30,11 +30,13 @@ import (
 )
 
 type objectAPI struct {
-	storage StorageAPI
+	Storage StorageAPI
```

Comment:
how come you are using caps for Storage here? what's our convention of using caps or not for variable names? (so that we can be consistent) Also is it a standard practice to allow caps in main package (as exporting types from main package is not meaningful)